presto-main/src/main/java/io/prestosql/cost/PlanCostEstimate.java

```diff
@@ -95,15 +106,16 @@ public double getNetworkCost()
      */
     public boolean hasUnknownComponents()
     {
-        return isNaN(cpuCost) || isNaN(memoryCost) || isNaN(networkCost);
+        return isNaN(cpuCost) || isNaN(maxMemory) || isNaN(maxMemoryWhenOutputting) || isNaN(networkCost);
     }
 
     @Override
     public String toString()
     {
         return toStringHelper(this)
                 .add("cpu", cpuCost)
-                .add("memory", memoryCost)
+                .add("memory", maxMemory)
+                // maxMemoryWhenOutputting is not that useful in toString
```

Comment:
because it's not taken into account when comparing cost to choose the best plan. So, anyone debugging a cost-based rule, will not benefit from this field in toString.